File: Structural/Decorator/elements/printed_tshirt_decorator.py
from .product_protocol import ProductProtocol

class PrintedTShirtDecorator(ProductProtocol):
   
    def __init__(self, product: ProductProtocol) -> None:
        self.product = product
        
    def get_price(self) -> int:
        return self.product.get_price() + 10
        

    def get_name(self) -> str:
        return self.product.get_name() + ' (printed)'

File: src/structural/decorator/typescript/classes/PrintedTshirtDecorator.ts
import { ProductProtocol } from './interfaces/ProductProtocol';

export class PrintedTShirtDecorator implements ProductProtocol {
  constructor(private product: ProductProtocol) {}

  get price(): number {
    return this.product.price + 10;
  }
  get name(): string {
    return this.product.name + ' (printed)';
  }
}
